devito/passes/clusters/misc.py

```diff
@@ -95,7 +95,11 @@ def callback(self, clusters, prefix):
 
             properties = c.properties.filter(key)
 
-            lifted.append(c.rebuild(ispace=ispace, properties=properties))
+            # Lifted scalar clusters cannot be guarded
+            # as they would not be in the scope of the guarded clusters
+            guards = {} if c.guards and c.is_scalar else c.guards
```

Comment:
explicit if's as by convention, also, you can drop the first part "if c.guards"

so:

```
if c.is_scalar:
    guards = {}
else:
    guards = c.guards
```

Reply:
Done